### PR TITLE
fix(training): nan mask weights gather for plots

### PR DESCRIPTION
**Pull request by japols**
## Description
Avoid gathering unresolved `nan_mask_weight` scaler with `grid_dim=1`.

## What problem does this change solve?
Fixes an bug where PlotLoss attempted to gather a nan_mask_weight scalar that wasn’t actually sharded but simply constant across the dimension (shape 1).

##  Additional notes ##

***As a contributor to the Anemoi framework, please ensure that your changes include unit tests, updates to any affected dependencies and documentation, and have been tested in a parallel setting  (i.e., with multiple GPUs). As a reviewer, you are also responsible for verifying these aspects and requesting changes if they are not adequately addressed. For guidelines about those please refer to https://anemoi.readthedocs.io/en/latest/***

By opening this pull request, I affirm that all authors agree to the [Contributor License Agreement.](https://github.com/ecmwf/codex/blob/main/Legal/contributor_license_agreement.md)

### CODE REVIEW

File: training/src/anemoi/training/diagnostics/callbacks/plot.py [hunk 1]
```diff
@@ -918,7 +918,8 @@ def on_validation_batch_end(
 
             self.loss = copy.deepcopy(pl_module.loss)
 
-            if hasattr(self.loss.scaler, "nan_mask_weights"):
+            # gather nan_mask_weights along grid dim if necessary, don't gather if grid dim is broadcastable
+            if hasattr(self.loss.scaler, "nan_mask_weights") and self.loss.scaler.nan_mask_weights.shape[-2] != 1:
```
Comment (anaprietonem) [inline training/src/anemoi/training/diagnostics/callbacks/plot.py]: @japols what do you mean here with broadcastable? Is this related to having sharding activated, probably could be good to clarify for users

**Comment by sahahner** [comment]
That is indeed a good point. We can remove the scaler for the lam and stretched grid scalers. 
Nevertheless, the general idea is to also define scalers that are not necessarily used (for example, the tendency scalers). Since the nan mask scaler is well tested for the global model, and I know users (not only myself :) ) who already use it, I would like to leave it in [global.yaml](https://github.com/ecmwf/anemoi-core/blob/main/training/src/anemoi/training/config/training/scalers/global.yaml).

**Comment by mchantry** [comment]
> Does it make sense to also comment out the `nan_mask_weights` in the scalers config if it is no longer being used?

@havardhhaugen there is a style at present to create the dictionary irrespective of use, and then just choose which ones to use in the training config. What you propose is reasonable but would break this style.

**Comment by anaprietonem** [comment]
PR missing another fix since output mask in on cpu when using plots as reported by Fernando (AEMET) - Jan planning to fix this in this PR and then we can finish review and merge

**Comment by anaprietonem** [comment]
@icedoom888  this is related to the issue you reported https://github.com/ecmwf/anemoi-core/issues/603 - could you please test this branch??

I run a simple LAM test with sharding and plots https://mlflow.ecmwf.int/#/experiments/264/runs/94dde82af22b4dc3a5ab5c4b7ac3d589
